### ISSUE TITLE
Provide wrapper to allow personalities to block on `Add` until leaf is integrated

### ISSUE THREAD

**Issue by mhutchinson** [transparency-dev/trillian-tessera#285]
We currently have an API that allows leaves to be added and blocked on until the leaf is sequenced. We need some kind of wrapper that extends this to allow personalities to block until the leaf is also integrated.

**Comment by mhutchinson** [comment]
Duplicate of #193.
